zephyr/lib/cpu.c

```diff
@@ -239,29 +206,24 @@ int cpu_enable_core(int id)
 
 int cpu_enable_secondary_core(int id)
 {
-	/*
-	 * This is an open-coded version of zephyr/kernel/smp.c
-	 * z_smp_start_cpu(). We do this, so we can use a customized
-	 * secondary_init() for SOF.
-	 */
-
 	if (arch_cpu_active(id))
 		return 0;
 
 #if ZEPHYR_VERSION(3, 0, 99) <= ZEPHYR_VERSION_CODE
-	z_init_cpu(id);
-#endif
-
-	atomic_clear(&start_flag);
-	atomic_clear(&ready_flag);
-
-	arch_start_cpu(id, z_interrupt_stacks[id], CONFIG_ISR_STACK_SIZE,
-		       secondary_init, &start_flag);
-
-	while (!atomic_get(&ready_flag))
-		k_busy_wait(100);
+	/* During kernel initialization, the next pm state is set to ACTIVE. By checking this
```

Comment:
Should we remove the ZEPHYR_VERSION check? Afer this change, we won't support older Zephyr 3.0 anymore in SOF main, so might as well drop this ifdef.

Reply:
sure, thanks